src/main/cpp/loggingevent.cpp

```diff
@@ -297,16 +302,19 @@ LoggingEvent::KeySet LoggingEvent::getPropertyKeySet() const
 }
 
 
-const LogString LoggingEvent::getCurrentThreadName()
+const LogString& LoggingEvent::getCurrentThreadName()
 {
 #if APR_HAS_THREADS
-	LOG4CXX_THREAD_LOCAL LogString thread_name;
-
-	if ( thread_name.size() )
+	thread_local LogString thread_id_string;
```

Comment:
This is probably going to fail if the compiler does not have `thread_local` support(hence why the `LOG4CXX_THREAD_LOCAL` macro existed).  Of course if `thread_local` does not exist, we can't return the `LogString` by reference since it would be declared on the stack and thus go away when the method returns.

Although cppreference does show that C++Builder [supports thread-local storage](https://en.cppreference.com/w/cpp/compiler_support/11)... do we need to test that and see if it actually does even if it claims it doesn't?